### DISCUSSION TITLE
Override default 'p' actions to always pasted from system clipboard.

### DISCUSSION THREAD

**Discussion by kelvinchincc** [zed-industries/zed#19818] [Feature Requests]
I'm trying to configure zed's vim keybinding to have both y and p to only copy or paste from the system clipboard, while others such as dd and ciw remains as internal register. However, I have no idea how to make p always paste from system clipboard as y is simply set the "use_system_clipboard" value in settings to on yank.

I have tried the following keymap configuration but it does not work as expected as I used to with .vimrc file. It will just override the p key and now waiting the last keybind forever, which will not happend at all and the vim command keys shown like this:
<img width="227" alt="image" src="https://github.com/user-attachments/assets/a8ba3e38-3d36-4c57-b777-7bf19abead11">

```json
[
  {
    "context": "Editor && vim_mode == normal && !menu",
    "bindings": {
      "p": ["workspace::SendKeystrokes", "\" + p"]
    }
  }
]
```

Is there anyway to configure zed to tempory ignore the custom keymaps or I have missed something in the keymap configuration file (eg: there's different configs that did similar thing)?

Btw I know that easier to just map it to ["workspace::SendKeystrokes": "a cmd-v escape"] but that's works completly different behaviour compare to vim as this issue (More configurable use_system_clipboard of vim mode #11313 ) points out.

**Comment by kelvinchincc** [comment]
Update: I'm currently have a very hacky way by overiding the default to other keybinds and remap p and P to it. It work as what I expected at version v0.158.2 as the following, but I'm not sure if there's any better ways?

```json
[
  {
    "context": "Editor && vim_mode == normal && !menu",
    "bindings": {
      "space p": "vim::Paste",
      "space P": ["vim::Paste", { "before": true }],
      "p": ["workspace::SendKeystrokes", "\" + space p"],
      "P": ["workspace::SendKeystrokes", "\" + space P"]
    }
  }
]
```

**Reply by kelvinchincc** [reply]
Update: seems like we will have better way on doing this on the latest pre release v0.161.0 as the kemapping have been improved. Will try it again and thanks to the team on keep improving zed!

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
